base/setup/lib/setuplib.c

```diff
@@ -628,18 +628,24 @@ InitDestinationPaths(
     IN PPARTENTRY PartEntry)    // FIXME: HACK!
 {
     WCHAR PathBuffer[MAX_PATH];
-
-    //
-    // TODO: Check return status values of the functions!
-    //
+    NTSTATUS Status;
 
     /* Create 'pSetupData->DestinationRootPath' string */
     RtlFreeUnicodeString(&pSetupData->DestinationRootPath);
-    RtlStringCchPrintfW(PathBuffer, ARRAYSIZE(PathBuffer),
-            L"\\Device\\Harddisk%lu\\Partition%lu\\",
-            DiskEntry->DiskNumber,
-            PartEntry->PartitionNumber);
-    RtlCreateUnicodeString(&pSetupData->DestinationRootPath, PathBuffer);
+    Status = RtlStringCchPrintfW(PathBuffer, ARRAYSIZE(PathBuffer),
+                     L"\\Device\\Harddisk%lu\\Partition%lu\\",
+                     DiskEntry->DiskNumber,
+                     PartEntry->PartitionNumber);
+
+    /* Check the condition status of RtlStringCchPrintfW() */
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT("RtlStringCchPrintfW() failed with status 0x%08lx\n", Status);
+        return Status;
+    }
+
+    /* Status assignment to Run-Time function returning boolean values (such values are of BOOLEAN type) */
+    Status = RtlCreateUnicodeString(&pSetupData->DestinationRootPath, PathBuffer) ? STATUS_SUCCESS : STATUS_NO_MEMORY;
```

Comment:
You're assigning a status but not checking it.

Reply:
Ugh, how are you going to check it if RtlCreateUnicodeString returns BOOLEAN?

Reply:
Ah yes, what @ThFabba wanted to say is that you don't do a:
```c
if (!NT_SUCCESS(Status))
{
    /* Display error here */
    /* Possibly deallocate memory allocated previously in this function */
    return Status;
}
```
after this call.

Reply:
@HBelusca Again, `RtlCreateUnicodeString()` returns a BOOLEAN value which means you don't simply use the conventional NT "return-error-if-on-fail" like doing in this snippet of code above. (which for this reason I put `? STATUS_SUCCESS : STATUS_NO_MEMORY` which is based on the boolean values which reflect these STATUS values).

Or are you telling me that it'll work in this way even if I do `!NT_SUCCESS(Status)` because of `? STATUS_SUCCESS : STATUS_NO_MEMORY`? If so, then I misunderstood Faber's point. Apologies.

Reply:
> Or are you telling me that it'll work in this way even if I do `!NT_SUCCESS(Status)` because of `? STATUS_SUCCESS : STATUS_NO_MEMORY`? If so, then I misunderstood Faber's point. Apologies.

Since you get a valid Status from `Status = function(...) ? STATUS_SUCCESS : STATUS_NO_MEMORY`, you can then follow that by the `if (!NT_SUCCESS(Status)) { ... }` block.

Reply:
Ah, that indeed makes sense. Thank you for your explanation Hermes, now it's more understandable!